### ISSUE TITLE
[Feature] Cell Boundary Expansion in DAPI Stained Fluorescence Images

### ISSUE THREAD

**Issue by guoyang-github** [gustaveroussy/sopa#130]
Hi,
Firstly, I want to express my gratitude for developing such a useful tool for efficient cell segmentation.
However, I've encountered a limitation in the current version of SOPA. The tool does an excellent job of segmenting cell nuclei for DAPI image, but it lacks the functionality to expand the segmented nuclei boundaries to simulate the cell membrane. This feature is crucial for representing the entire cell, not just the nucleus.
I would like to propose the addition of a new feature that allows users to expand the segmented cell nuclei by a specified distance to estimate the cell membrane boundary. 
Here's a suggested outline for the functionality:
    1. A option to select the distance by which the cell boundary should be expanded,
    2. The ability to apply this expansion uniformly to all segmented nuclei, and ensure no overlap between adjacent cells,
    3. An option to visualize the expanded boundaries in a separate layer for better analysis.

Thank you for considering this feature request.

**Comment by quentinblampey** [comment]
Hello @guoyang-github,
This feature actually already exists: there is a parameter called `expand_radius_ratio` in the aggregation. This way, it will expand the cell before counting the transcripts or averaging the channel intensities.

If you want to visualize this radius expansion, you can also update directly your SpatialData object. In this example, I expand the radius of the `'cellpose_boundaries'` by 10 pixels:
```python
sdata['cellpose_boundaries'].geometry = sdata['cellpose_boundaries'].buffer(10)
```

Let me know if you need more details or if you have additional questions!

**Comment by guoyang-github** [comment]
Hello @quentinblampey ,
Thanks for your response. 
I noticed that the method mentioned above can expand the cell boundaries, but this may result in overlapping of the cell boundaries, which would prevent the accurate assignment of transcripts to individual cells. 
I think a more suitable approach would be to expand the nucleus boundaries by a specified distance or until they encounter another cell boundary.

Thanks,

**Comment by quentinblampey** [comment]
Yes, true, this would be better. I note this as a feature request, and I'll work on this!

**Comment by NadineBestard** [comment]
As an additional comment to this feature request, for overlapping transcripts, it would be nice to use the GEX to refine the assignment to one cell or another.

**Comment by quentinblampey** [comment]
I added a `no_overlap: bool` argument when performing the radius expansion, which will be available in the next release, although still experimental. You can already test this using the`main` branch if you want

> it would be nice to use the GEX to refine the assignment to one cell or another.

Can you confirm that what you want is the following: when one bin is inside multiple expanded-nuclei, we assign it only to the nucleus whose cell expression is the most similar?

**Comment by NadineBestard** [comment]
yes, this is what I meant! Sorry my sentence did not make sense, I'm glad you got me haha

**Comment by quentinblampey** [comment]
Great! This feature will also be in the next release. But it's still experimental, so I'll be happy to have your feedback!
